addon/ng2/commands/generate.ts

```diff
@@ -40,6 +44,45 @@ const GenerateCommand = EmberGenerateCommand.extend({
     };
     
     return EmberGenerateCommand.prototype.beforeRun.apply(this, arguments);
+  },
+
+  run: function(commandOptions, rawArgs) {
+    var blueprintName = rawArgs[0];
+
+    if (!blueprintName) {
+      return Promise.reject(new SilentError('The `ember generate` command requires a ' +
+                                            'blueprint name to be specified. ' +
+                                            'For more details, use `ember help`.'));
+    }
+
+    var Task = this.tasks.GenerateFromBlueprint;
+    var task = new Task({
+      ui: this.ui,
+      analytics: this.analytics,
+      project: this.project,
+      testing: this.testing,
+      settings: this.settings
+    });
+
+    var taskArgs = {
+      args: rawArgs
+    };
+
+    if (this.settings && this.settings.usePods && !commandOptions.classic) {
+      commandOptions.pod = !commandOptions.pod;
+    }
+
+    var taskOptions = _.merge(taskArgs, commandOptions || {});
+
+    if (this.project.initializeAddons) {
+      this.project.initializeAddons();
+    }
+
+    if (rawArgs[0] === 'component') {
+      return Promise.all([task.run(taskOptions), importIntoAppModule(rawArgs, this.project)])
```

Comment:
you probably want to wait for `task` to be done, in case there's an error. This should be:

```typescript
return task.run(taskOptions)
  .then(() => importIntoAppModule(rawArgs, this.project));
```